packages/babel-plugin-proposal-class-properties/src/index.js

```diff
@@ -90,11 +90,37 @@ export default function(api, options) {
         const staticNodes = [];
         let instanceBody = [];
 
+        const ClassFieldDefinitionEvaluationTDZ = (classRef, state) => ({
```

Comment:
This visitor can be moved to the top of the plugin factory function (alongside with [`bareSupers`](https://github.com/maurobringolf/babel/blob/9488e10716ae7f4bec5c7fc51f80bf1ad6872530/packages/babel-plugin-proposal-class-properties/src/index.js#L8) and [`referenceVisitor`](https://github.com/maurobringolf/babel/blob/9488e10716ae7f4bec5c7fc51f80bf1ad6872530/packages/babel-plugin-proposal-class-properties/src/index.js#L16)), so we avoid redefining this function for every class and calling it for every static computed property. You can pass `classRef` and `addHelper` using the state. e.g.

```diff
           ReferencedIdentifier(path) {
-            if (classRef === path.scope.getBinding(path.node.name)) {
-              const classNameTDZError = state.addHelper("classNameTDZError");
+            if (this.classRef === path.scope.getBinding(path.node.name)) {
+              const classNameTDZError = this.file.addHelper(
+                "classNameTDZError",
+              );
               const throwNode = t.callExpression(classNameTDZError, [
```
```diff
           if (!computedPath.get("key").isConstantExpression()) {
-            computedPath.traverse(
-              ClassFieldDefinitionEvaluationTDZ(
-                path.scope.getBinding(ref.name),
-                state,
-              ),
-            );
+            computedPath.traverse(classFieldDefinitionEvaluationTDZVisitor, {
+              classRef: path.scope.getBinding(ref.name),
+              file: this.file,
+            });
             const ident = path.scope.generateUidIdentifierBasedOnNode(
```